disperser/apiserver/disperse_blob_v2.go

```diff
@@ -17,9 +17,27 @@ import (
 	"github.com/Layr-Labs/eigenda/encoding"
 	"github.com/Layr-Labs/eigenda/encoding/rs"
 	gethcommon "github.com/ethereum/go-ethereum/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-func (s *DispersalServerV2) DisperseBlob(ctx context.Context, req *pb.DisperseBlobRequest) (*pb.DisperseBlobReply, error) {
+func (s *DispersalServerV2) DisperseBlob(
+	ctx context.Context,
+	req *pb.DisperseBlobRequest,
+) (*pb.DisperseBlobReply, error) {
+	reply, st := s.disperseBlob(ctx, req)
+	api.LogResponseStatus(s.logger, st)
+	if st != nil {
+		// nolint:wrapcheck
+		return reply, st.Err()
+	}
+	return reply, nil
+}
+
+func (s *DispersalServerV2) disperseBlob(
+	ctx context.Context,
+	req *pb.DisperseBlobRequest,
+) (*pb.DisperseBlobReply, *status.Status) {
```

Comment:
What is the reasoning being switching the return types of all these internal methods to `*status.Status`?

It seems like we could accomplish the same thing if we left the return types as errors, and then log the errors within `DisperseBlob`? Are we gaining anything by wrapping these grpc errors up into statuses, just to turn around and unpack them back into errors in `DisperseBlob`?